src/HyperLogLog.net/HyperLogLog.net/HyperLogLog.cs

```diff
@@ -80,11 +92,14 @@ public int GetCount()
                 else
                     dv = (int)(M * Math.Log((double)M / v));
             }
-            else if (dvEstimate <= (Pow2_32 * 1 / 30))
+            else if (dvEstimate <= (Pow2_32 * 1 / 30))  // 143,165,576
                 dv = (int)dvEstimate;
             else
             {
-                dv = (int)(Math.Pow(-2, 32) * Math.Log(1 - dvEstimate / Pow2_32));
+                if (dvEstimate <= 1E9)
+                    dv = (int)(-Pow2_32 * Math.Log(1 - dvEstimate / Pow2_32));
```

Comment:
the formula in the article is -2^32, not (-2)^32. if see it the latter way, it gives wrong negative results becase log is always negative